pkg/config/controller_config.go

```diff
@@ -88,3 +99,38 @@ func configInit(metadata EC2Metadata) error {
 
 	return nil
 }
+
+// try to find cluster name, search in env then in ec2 instance tags
+// returns empty string if not found
+func getClusterName(sess *session.Session) (string, error) {
+	cn := os.Getenv(CLUSTER_NAME)
+	if cn != "" {
+		return cn, nil
+	}
+	// fallback to ec2 instance tags
+	meta := ec2metadata.New(sess)
+	doc, err := meta.GetInstanceIdentityDocument()
+	if err != nil {
+		return "", err
+	}
+	instanceId := doc.InstanceID
+	region, err := meta.Region()
+	if err != nil {
+		return "", err
+	}
+	ec2Client := ec2.New(sess, &aws.Config{Region: aws.String(region)})
+	tagReq := &ec2.DescribeTagsInput{Filters: []*ec2.Filter{{
+		Name:   aws.String("resource-id"),
+		Values: []*string{aws.String(instanceId)},
+	}}}
+	tagRes, err := ec2Client.DescribeTags(tagReq)
+	if err != nil {
+		return "", err
+	}
+	for _, tag := range tagRes.Tags {
+		if *tag.Key == "aws:eks:cluster-name" {
+			return *tag.Value, nil
+		}
+	}
+	return "", nil
```

Comment:
should we instead want to fail here? It seems like we will not have the cluster name and user wouldn't know that now there are these edge cases they will run into

Reply:
yeah, it seems everyone is moving towards cluster-name must have variable.